vms/platformvm/network/gossip.go

```diff
@@ -92,49 +92,63 @@ type gossipMempool struct {
 	bloom *gossip.BloomFilter
 }
 
-func (g *gossipMempool) Add(tx *txs.Tx) error {
-	txID := tx.ID()
-	if _, ok := g.Mempool.Get(txID); ok {
-		return fmt.Errorf("tx %s dropped: %w", txID, mempool.ErrDuplicateTx)
+func (g *gossipMempool) Add(txns ...*txs.Tx) []error {
+	errs := make([]error, len(txns))
+	hasErr := false
+	for i, tx := range txns {
+		txID := tx.ID()
+		if _, ok := g.Mempool.Get(txID); ok {
+			errs[i] = fmt.Errorf("tx %s dropped: %w", txID, mempool.ErrDuplicateTx)
+			hasErr = true
+			continue
+		}
+
+		if errs[i] = g.Mempool.GetDropReason(txID); errs[i] != nil {
+			// If the tx is being dropped - just ignore it
+			//
+			// TODO: Should we allow re-verification of the transaction even if it
+			// failed previously?
+			hasErr = true
+			continue
+		}
+
+		if errs[i] = g.txVerifier.VerifyTx(tx); errs[i] != nil {
+			g.Mempool.MarkDropped(txID, errs[i])
+			hasErr = true
+			continue
+		}
+
+		if errs[i] = g.Mempool.Add(tx); errs[i] != nil {
+			g.Mempool.MarkDropped(txID, errs[i])
+			hasErr = true
+			continue
+		}
+
+		g.lock.Lock()
+
+		g.bloom.Add(tx)
+		var reset bool
+		reset, errs[i] = gossip.ResetBloomFilterIfNeeded(g.bloom, g.Mempool.Len()*bloomChurnMultiplier)
+		if errs[i] != nil {
+			hasErr = true
+			continue
```

Comment:
We should add tests for multiple txs - I think this can continue to the next iteration with the lock held.

Would it make sense to keep the single tx addition as a method so that we can keep using `defer g.lock.Unlock()`?